### PR TITLE
Total row for UCR

### PR DESCRIPTION
**Pull request by nickpell**
For Abt UCR.  Best read commit by commit

@NoahCarnahan @czue cc: @gcapalbo

### CODE REVIEW

**Comment by czue** [comment]
looks good, but you should know better than to write UCR code without tests :stuck_out_tongue_winking_eye:

**Comment by NoahCarnahan** [comment]
does would be nice too :)

File: corehq/apps/userreports/reports/data_source.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import datetime
 from django.utils.datastructures import SortedDict
+from django.utils.translation import ugettext
```
Comment (dimagimon) [inline corehq/apps/userreports/reports/data_source.py]: F401 'ugettext' imported but unused

**Comment by nickpell** [comment]
updated tests and docs

File: corehq/apps/userreports/reports/util.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from django.utils.translation import ugettext
+
+
+def get_total_row(data, aggregation_columns, column_configs):
+    total_row = [0] * len(column_configs)
+    for i, column_config in enumerate(column_configs):
+        if column_config.calculate_total:
+            total_row[i] = reduce(
```
Comment (czue) [inline corehq/apps/userreports/reports/util.py]: sorry to nitpick, but i still feel like this function was designed to handle edge cases involving datatypes that aren't tested in the tests you added. possible to pull it out and exercise those?

**Comment by nickpell** [comment]
@czue let me know if you think https://github.com/dimagi/commcare-hq/commit/a75c9288b79c35816e0495a25b46e7efc1d94f50 does the job.  I feel like that lambda function is more of an implementation detail than an essential function (I had originally considered implementing this without defining a lambda) so I chose to provide an input with the relevant test coverage without actually pulling out the function.

**Comment by czue** [comment]
@nickpell I think that works. I often try to write the simplest test for the bit of code that I want to exercise which would be pulling it out and writing a few one liners (the test you added was closer to 50 lines) but that's subjective enough for me not to make a stink about it

**Pull request merged**